Marble Blast Platinum/platinum/client/ui/ExitGameDlg.gui

```diff
@@ -302,30 +545,157 @@ function ExitGameDlg::onWake(%this) {
 		%awesomeTitle = "<shadow:1:1><shadowcolor:000000bf><color:FF4444>Awesome";
 	}
 
-	%parText      = %parTitle;
-	%goldText     = %goldTitle;
-	%platinumText = %platinumTitle;
-	%ultimateText = %ultimateTitle;
-	%awesomeText  = %awesomeTitle;
+	%recordTitle   = "<shadow:1:1><shadowcolor:000000bf><color:0060f0>Record";
+	%eggRecordTitle = "<shadow:1:1><shadowcolor:000000bf><color:4580ff>Egg Record";
+
+	//Get the world record
+	%record = false;
+	if (lb()) {
+		
+		%cache = PlayMissionGui.globalScoreCache[PlayMissionGui.getMissionInfo().id];
+		if (isObject(%cache)) {
+			%scores = %cache.scores;
+			if (%scores.getSize()) {
+				%record = %scores.getEntry(0).score;
+				%recordType = %scores.getEntry(0).score_type;
+				%recordLabel = (%recordType $= "time" ? formatTime(%record) : formatScore(%record));
+			}
+		}
 
-	//Par can have both time and score
-	if (ClientMode::callbackForMission(%info, "timeMultiplier", 1) > 0) {
-		if (%parTimeLabel !$= "N/A") {
-			%text = %text NL "<just:left><spush>" @ %parText @ ":<spop><just:right>" @ %parTimeLabel;
+		%cache = PlayMissionGui.globalScoreModeCache[PlayMissionGui.getMissionInfo().id]; 
+		if (isObject(%cache)) {
+			%scores = %cache.getFieldValue("egg").scores; // Egg scores
+			if (%scores.getSize()) {
+				%eggRecordLabel = formatTime(%scores.getEntry(0).time);
+			}
 		}
 	}
-	if (%parScoreLabel !$= "N/A") {
-		%text = %text NL "<just:left><spush>" @ %parTitle SPC "Score:<spop><just:right>" @ %parScoreLabel;
+	%challengeTitles = "";
+	%challengeTimes = "";
+
+	%info = PlayMissionGui.getMissionInfo();
+	%pbType = getField($hs[0], 0);
+	%pbLabel = %pbType == 0 ? formatTime(getField($hs[0], 1)) : formatScore(getField($hs[0], 1));
+	%pbColor = getScoreFormatting($hs[0], %info, "", (%record == getField($hs[0], 1))? 1 : 2); // Technically no record (offline) and score of 0 counts as a record, but PB should not show in that case.
+	%pbEggLabel = formatTime(getField($hs["egg"], 0));
+
+	%pbTitle = "<shadow:1:1><shadowcolor:000000bf>Personal Best";
+	%pbEggTitle = "<shadow:1:1><shadowcolor:000000bf><color:4580ff>Egg Best";
+
+	if ((%pbType == 0 && getSubStr(%pbLabel, 0, 5) $= "99:59") || (%pbType == 1 && getField($hs[0], 1) == 0)) {
```

Comment:
Technically speaking, 99:59.999 can be your best time on a level. You should compare to the default score instead (see getBestTimes() in client/scripts/game.cs)